theforeman.org/pipelines/lib/packaging.groovy

```diff
@@ -0,0 +1,455 @@
+
+        dependencies.add([
+            type: 'dependency',
+            name: dependency_split[2],
+            path: "${dependency_split[0]}/${dependency_split[1]}/${dependency_split[2]}",
+            operating_systems: dependency_split[1]
+        ])
+    }
+
+    for(plugin in changed_plugins) {
+        plugin_split = plugin.split('/')
+
+        plugins.add([
+            type: 'plugin',
+            name: plugin_split[1],
+            path: "${plugin_split[0]}/${plugin_split[1]}",
+            operating_systems: ['bionic']
+        ])
+    }
+
+    return cores + dependencies + plugins
+}
+
+def query_rpmspec(specfile, queryformat) {
+    result = sh(returnStdout: true, script: "rpmspec -q --srpm --undefine=dist --undefine=foremandist --queryformat=${queryformat} ${specfile}").trim()
+    return result
+}
+
+def repoclosure(repo, dist, version) {
+    version = version == 'nightly' ? 'develop' : version
+    git_repo = repo == 'pulpcore' ? 'pulpcore-packaging' : 'foreman-packaging'
+    ws(dist) {
+        dir('packaging') {
+            git url: "https://github.com/theforeman/${git_repo}", branch: "rpm/${version}", poll: false
+            setup_obal()
+            obal(
+                action: 'repoclosure',
+                packages: "${repo}-repoclosure-${dist}"
+            )
+        }
+    }
+}
+
+def repoclosures(repo, releases, version) {
+    def closures = [:]
+
+    releases.each { release ->
+        closures[release] = { repoclosure(repo, release, version) }
+    }
+
+    return closures
+}
+
+def build_deb_package_steps(packages_to_build, version, repoowner = 'theforeman', pull_request = false) {
+    def steps = [
+        configure_pbuilder: [:],
+        setup_sources: [:],
+        executes: [:],
+        stage_packages: [:]
+    ]
+
+    for(int i = 0; i < packages_to_build.size(); i++) {
+        def index = i
+        def pkg_to_build = packages_to_build[index]
+        def project = pkg_to_build['name']
+
+        echo "${pkg_to_build}"
+
+        for(int j = 0; j < pkg_to_build['operating_systems'].size(); j++) {
+            def os_index = j
+            def os = pkg_to_build['operating_systems'][j]
+            def execute_pbuilder_dir
+
+            steps.configure_pbuilder["${os}-${project}"] = {
+                configure_pbuilder(project, os, version, repoowner)
+            }
+
+            steps.setup_sources["${os}-${project}"] = {
+                execute_pbuilder_dir = setup_sources_deb(pkg_to_build['type'], project, os, version, repoowner, pull_request)
+            }
+
+            steps.executes["${os}-${project}"] = {
+                execute_pbuilder(execute_pbuilder_dir, os)
+            }
+
+            steps.stage_packages["${os}-${project}"] = {
+                stage_debian_packages(pkg_to_build['type'], project, os, version, repoowner, pull_request)
+            }
+        }
+    }
+
+    return steps
+}
+
+def configure_pbuilder(project, os, version, repoowner) {
```

Comment:
This depends on the outcome of https://github.com/theforeman/jenkins-jobs/pull/95/files and https://github.com/theforeman/foreman-infra/pull/1642